NewHaven/src/Square.cpp

```diff
@@ -7,30 +7,54 @@
 
 using namespace std;
 
-Square::Square(){
-    position = 0;
-    tile = 0;
+Square::Square() : visitedResource{ new bool[4]{ false, false, false, false } }, rootConnected{ new bool[4]{ false, false, false, false } }
+{
+    position = nullptr;
+    tile = nullptr;
     isVisited = new bool(false);
     isPlayed = new bool(false);
 }
 
+Square::Square(const Square& square) : visitedResource{ new bool[4] }, rootConnected{ new bool[4] }
+{
+    position = new int(*square.position);
+    if(square.tile != nullptr)
+        tile = new HarvestTile(*square.tile);
+    else
+        tile = nullptr;
+    isVisited = new bool(*square.isVisited);
+    isPlayed = new bool(*square.isPlayed);
+
+    for (std::int_fast8_t i = 0; i < 4; i++)
+    {
+        visitedResource[i] = square.visitedResource[i];
+        rootConnected[i] = square.rootConnected[i];
+    }
+}
+
+Square::~Square() = default;
+
 void Square::setPosition(int* position){
-    this->position = position;
+    int *cpy = new int(*position);
+    this->position = new int(*cpy);
+    delete cpy;
+    delete position;
 }
```

Comment:
Just pitching in my thought.
`
void Square::setPosition(const int& position){
    delete this->position;
    this->position = new int(position);
}
`
Deleting the old memory of position because it's to be replaced. Then we use the copy constructor of int() to assign position a copy of the argument. No need to make the change now anyways.

Reply:
Maybe we could even skip the extra steps and  use just `*this->position = position;`.

Reply:
this->position = new int(position); is sufficient idk why i put two copies its redundant. 
 delete this->position is useless, because we are using a copy constructor and position is set to a nullptr by default so there is nothing to delete.